### ISSUE TITLE
conda create causes: AttributeError: 'dict' object has no attribute 'add'

### ISSUE THREAD

**Issue by twiecki** [conda/conda#3060]
Trying to clone the root environment with:
```
conda create --clone root --copy -n myenv2
```
Gives:
```
Traceback (most recent call last):
  File "/opt/miniconda/bin/conda", line 6, in <module>
    sys.exit(main())
  File "/opt/miniconda/lib/python2.7/site-packages/conda/cli/main.py", line 120, in main
    exit_code = args_func(args, p)
  File "/opt/miniconda/lib/python2.7/site-packages/conda/cli/main.py", line 130, in args_func
    exit_code = args.func(args, p)
  File "/opt/miniconda/lib/python2.7/site-packages/conda/cli/main_create.py", line 58, in execute
    install(args, parser, 'create')
  File "/opt/miniconda/lib/python2.7/site-packages/conda/cli/install.py", line 223, in install
    clone(args.clone, prefix, json=args.json, quiet=args.quiet, index_args=index_args)
  File "/opt/miniconda/lib/python2.7/site-packages/conda/cli/install.py", line 88, in clone
    index_args=index_args)
  File "/opt/miniconda/lib/python2.7/site-packages/conda/misc.py", line 360, in clone_env
    force_extract=False, index_args=index_args)
  File "/opt/miniconda/lib/python2.7/site-packages/conda/misc.py", line 109, in explicit
    channels.add(channel)
AttributeError: 'dict' object has no attribute 'add'

```

conda info:
```
             platform : linux-64
        conda version : 4.1.8
    conda-env version : 2.5.1
  conda-build version : 1.18.1
       python version : 2.7.12.final.0
     requests version : 2.10.0
     root environment : /opt/miniconda  (read only)
  default environment : /opt/miniconda
     envs directories : /home/wiecki/.conda/envs
                        /home/wiecki/envs
                        /opt/miniconda/envs
        package cache : /home/wiecki/.conda/envs/.pkgs
                        /home/wiecki/envs/.pkgs
                        /opt/miniconda/pkgs
         channel URLs : https://repo.continuum.io/pkgs/free/linux-64/
                        https://repo.continuum.io/pkgs/free/noarch/
                        https://repo.continuum.io/pkgs/pro/linux-64/
                        https://repo.continuum.io/pkgs/pro/noarch/
          config file : None
         offline mode : False
    is foreign system : False
```

**Comment by Evfro** [comment]
Same problem on Windows. Also encountered https://github.com/conda/conda/issues/3009

**Comment by HugoTian** [comment]
On Mac, it also has this bug. In the source code, channels is initialised as {}, and dict has no add() function. I change it to channels = () at line 51, and every thing is fine. 

Thanks for pointing out the bug

**Comment by HugoTian** [comment]
@kalefranz , this is a bug need to fix for conda 4.2 I think, and I will create a PR for that

**Comment by HugoTian** [comment]
PR #3078  will fix this one
